main/src/library/Fixpoint3/Interpreter.flix

```diff
@@ -0,0 +1,699 @@
+            let (db, _, _) = ctxt;
+            let src = lookupIndex(srcIndex, db);
+            let dst = lookupIndex(dstIndex, db);
+            BPlusTree.mergeWith(lub, src, dst)
+        case RamStmt.Swap(lhs, rhs) =>
+            let (db, _, _) = ctxt;
+            let temp = lookupIndex(lhs, db);
+            storeIndex(lookupIndex(rhs, db), lhs, db);
+            storeIndex(temp, rhs, db)
+        case RamStmt.Purge(index) => 
+            let (db, _, _) = ctxt;
+            let oldIndex = lookupIndex(index, db);
+            let newIndex = BPlusTree.emptyWithArityAndSearch(rc, usedArity(), BPlusTree.search(oldIndex));
+            storeIndex(newIndex, index, db)
+        case RamStmt.Seq(stmts) => 
+            Vector.forEach(evalStmt(rc, ctxt, expectedJoinSize, parLevel), stmts)
+        case RamStmt.Until(test, body) =>
+            if (evalBoolExps(ctxt, test)) {
+                ()
+            } else {
+                evalStmt(rc, ctxt, expectedJoinSize, parLevel, body);
+                evalStmt(rc, ctxt, expectedJoinSize, parLevel, stmt)
+            }
+        case RamStmt.Comment(_) => ()
+        // Imortant to note: all threads spawned on region `rc2` will have terminated before we exit `rc2`.
+        case RamStmt.Par(stmts) => region rc2 {
+            stmts |> Vector.forEach(innerStmt -> unchecked_cast({
+                let _: Unit = spawn unchecked_cast({
+                    let threadCtxt = makeCtxt(rc, ctxt);
+                    evalStmt(rc, threadCtxt, expectedJoinSize, parLevel, innerStmt)
+                } as _ \ IO) @ rc2; // spawn may not take an `r` effect, so we cast it to `IO`
+                ()
+            } as _ \ rc2)) // The `IO` effect is undesirable, so we cast it back to an `r` effect
+        }
+        case RamStmt.EstimateJoinSize(indexPos, writeTo, attr) =>
+            let prevRef = Ref.fresh(rc, None);
+            let dupRef = Counter.fresh(rc);
+            let sizeRef = Counter.fresh(rc);
+            let (db, _, _) = ctxt;
+            let relIndex = lookupIndex(indexPos, db);
+            relIndex |> BPlusTree.forEach(curr -> _ -> {
+                let isDup = 
+                    Ref.get(prevRef) |> 
+                    Option.exists(prev -> 
+                        attr |> Vector.forAll(i -> Vector.get(i, prev) == Vector.get(i, curr))
+                    );
+                if (isDup) {
+                    Counter.increment(dupRef)
+                } else ();
+                Ref.put(Some(curr), prevRef);
+                Counter.increment(sizeRef)
+            });
+            let size = Counter.get(sizeRef);
+            let dup = Counter.get(dupRef);
+            MutList.push((size, dup), Array.get(writeTo, getOrCrash(expectedJoinSize)))
+    }
+
+    ///
+    /// Evaluate `op`, under `ctxt` with at most `parLevel` allowed new branches `RelOp.Search`.
+    ///
+    def evalOp(rc: Region[r], ctxt: Context[r], parLevel: AllowedBranches, op: RelOp): Unit \ r = 
```

Comment:
I just noticed you call it "ctxt", in Flix we have many contexts and we always call them "ctx". Can we make that changes here too?